install python modules in Terminal when `python.installModulesInTerminal` enabled

sharon-wang opened the pull request:
- addresses https://github.com/posit-dev/positron/issues/5506
- adds a new command `python.installModulesInTerminal`, that, when enabled, forces python installations to occur in the Terminal
    - this option is disabled by default
- passes an otherwise unused cancellation token to the terminal service, which makes it so that the terminal execution must complete before the promise resolves
    - This might mean that we can opt to use the Terminal (and pass a cancellation token) in the places where we have the comment
    ```ts
            // Using a process to install modules avoids using the terminal service,
            // which has issues waiting for the outcome of the install.
    ```
    - https://github.com/posit-dev/positron/blob/54cbc06c099bc8910011bc31c3586fb1882d2fd2/extensions/positron-python/src/client/common/terminal/types.ts#L44

### QA Notes

When `python.installModulesInTerminal` is enabled, any flow that passes through the `installModule()` method should run the install command in a Terminal visible to the user. When the option is not enabled, python module installations should occur in the background, invisible to the user. 

#### Project Wizard
1. Create a Python or Jupyter project with Venv
2. One of the new project initialization tasks should kick off the `ipykernel` install in a Terminal

#### Command Prompt
1. Run the command `Python: Create Environment`
2. Start the newly created interpreter
3. Accept the `ipykernel` install
4. See `ipykernel` install in a Terminal

#### Select interpreter that does not have `ipykernel`
1. Select an interpreter from the Interpreter Dropdown or use the `Python: Select Interpreter` command that _does not_ already have `ipykernel` installed
2. Accept the `ipykernel` install
3. See `ipykernel` install in a Terminal

sharon-wang committed:
force python install to terminal

adds a new command `python.installModulesInTerminal`, that, when enabled, forces python installations to occur in the Terminal.

jmcphers commented on extensions/positron-python/package.json:
```diff
@@ -837,6 +837,12 @@
+                },
+                "python.installModulesInTerminal": {
+                    "default": false,
+                    "description": "%python.installModulesInTerminal.description%",
```
Nit: Since you're using markdown styling in here, consider specifying `markdownDescription` instead so you get nicer rendering.

jmcphers reviewed (dismissed):
LGTM; confirmed that turning on this new setting lets me see the installs in a Terminal.

seeM commented on extensions/positron-python/src/client/common/installer/moduleInstaller.ts:
```diff
@@ -44,7 +51,16 @@ export abstract class ModuleInstaller implements IModuleInstaller {
+        // python.installModulesInTerminal is a setting that allows the user to force modules to be
+        // installed in the Terminal. Usually, such installations occur in the background. However,
+        // for debugging, it can be helpful to see the Terminal output of the installation process.
+        const workspaceService = this.serviceContainer.get<IWorkspaceService>(IWorkspaceService);
```
There's also `IConfigurationService` that is typed, not critical at all.

sharon-wang replied:
For some reason, when using `IConfigurationService`, I keep getting the default `false` instead of the actual current user setting value. I've now removed the unused ConfigurationService definitions

seeM replied:
You have to also set the values here: https://github.com/posit-dev/positron/blob/fix-duplicate-notebook-progress-bars/extensions/positron-python/src/client/common/configSettings.ts#L316. I'm not really sure if it's worth it.

seeM reviewed (dismissed):
Code changes LGTM!

sharon-wang committed:
use `markdownDescription` for nicer rendering

sharon-wang committed:
fix moduleInstaller unit tests

sharon-wang committed:
apply formatting fixes

sharon-wang committed:
add back deleted newline

sharon-wang committed:
remove unused definitions for `installModulesInTerminal`

leftover from a previous attempt to use ConfigurationService instead of WorkspaceService

sharon-wang committed:
move `installModulesInTerminal` check to method
